src/main/httpCommandServer.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs';
+import http from 'http';
+import path from 'path';
+
+import Logging from '@/utils/logging';
+import paths from '@/utils/paths';
+
+type ServerState = {
+  user: string;
+  password: string;
+  port: number;
+  pid: number;
+}
+
+const console = Logging.server;
+const ServerPort = 6107;
+const ServerUsername = 'user';
+const ServerFileBasename = 'rd-engine.json';
+
+export default class HttpCommandServer {
+  protected server = http.createServer();
+  protected password = randomStr();
+  protected stateInfo: ServerState = {
+    user:     ServerUsername,
+    password: this.password,
+    port:     ServerPort,
+    pid:      process.pid,
+  };
+
+  async init() {
+    const statePath = path.join(paths.appHome, ServerFileBasename);
+
+    await fs.promises.writeFile(statePath,
+      JSON.stringify(this.stateInfo, undefined, 2),
+      { mode: 0o600 });
+    this.server.listen(ServerPort);
```

Comment:
@ericpromislow Can you please confirm that this will bind to `127.0.0.1` and not to `0.0.0.0`.

Due to the concerns raised by @AkihiroSuda we don't want to expose the port externally in the prototype implementation.

Reply:
I added the '127.0.0.1' field to `listen` and couldn't reach `192.168.205.1:6107` from the lima VM, but I could ping the IP. However I got the same 'Connection refused' error without specifying an address in the call to `listen`, so that's obviously not a complete test. Maybe someone with a better network setup than I have can verify this.